workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReaderWriterFactory.cs

```diff
@@ -1,12 +1,66 @@
-﻿using System;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core.MonoBehaviours;
+using Unity.Entities;
+using UnityEngine;
 
 namespace Improbable.Gdk.Core
 {
-    public static class ReaderWriterFactory
+    public class ReaderWriterFactory
     {
-        public static object CreateReaderWriter(uint componentId)
+        private const string ComponentIdAttributeNotFoundError
+            = "ReaderWriterCreator found with no Component ID attribute.";
+        private const string NoReaderWriterCreatorFoundForComponentIdError
+            = "No ReaderWriterCreator found for componentId.";
```

Comment:
"No ReaderWriterCreator found for the given ComponentId"

Reply:
Fixed!